crate2nix/templates/nix/crate2nix/default.nix

```diff
@@ -571,6 +573,40 @@ rec {
     then builtins.throw "strictDeprecation enabled, aborting: ${message}"
     else builtins.trace message value;
 
+  /* create a shell express for the given crates from the crateOverrides */
+  buildWorkspaceShell = crateOverrides: crates:
+    assert builtins.isAttrs crates;
+    assert builtins.isAttrs crateOverrides;
+    let
+      overrides =
+        lib.attrValues
+          (
+            lib.mapAttrs (n: v: v { })
+              (lib.filterAttrs (a: _: builtins.hasAttr a crates) crateOverrides)
+          );
+
+      # Extract environment variables (~ upper case attributes) from the overrides.
+      # This will throw if there are conflicting definitions of one environment variable.
```

Comment:
Two points:
  * I believe we should try to provide them is a working solution out of the box thus telling them the the setup they have doesn't work (without further manual work). If some environment variables is contradicting another variable that is probably a bug in the build system setup (or an "impurity" of sorts). You can't build a project that has two crates depending on different values of the same environment variable with a regular cargo build. That would be confusing to users.

  * I can totally agree that having a mode (e.g. `ignoreDuplicateEnvVars = true` or similar) would be desireable in some situations where the user is experienced. That would downgrade the `throw` into some sort of warning. The issue here, thus I don't like it being the default way, is that Nix usually produces a bit of output and after you started your shell you might not actually see that warning output without looking at it explicitly.

Reply:
`ignoreDuplicateEnvVars` is an awesome idea! I like it better than what I proposed because, as you wrote, a warning can be super easily missed. It would be even cooler if it wasn't a boolean but listed the variables to ignore. If it feels like over-engineering, ignore.